### DISCUSSION TITLE
Use something other than \q_stop in \evalWhole implementation?

### DISCUSSION THREAD

**Discussion by user202729** [lvjr/functional#12] [General]
Maybe make a custom quark for this purpose (also make it `\outer` to prevent the user accidentally grabbing the token)? I think `\q_stop` is a bit too commonly-used to use as a delimiter.

**Comment by lvjr** [comment]
I think nearly no users will use `\q_stop` in the argument, but yes it will be a little safer to replace it with `\q__fun_stop`.

**Comment by lvjr** [comment]
Outer macros are not used in LaTeX. I don't think it is a good idea to use it any more. Does `expl3` use it?

**Reply by user202729** [reply]
Up to you, then.

For reference I think it's used somewhere in the code that marks function as deprecated; although a few things e.g. `peek_analysis` functions also somewhat support them.

In this case I think it does help (avoid user accidentally grabbing the end token) and won't ever give error in normal usage (for all the "normal usage" cases I can think of).
